mathesar/models.py

```diff
@@ -0,0 +1,49 @@
+from django.db import models
+from sqlalchemy.orm import Session
+
+from mathesar.database.base import db_name, engine, metadata
+from mathesar.database.collections import DBCollection
+
+
+class DatabaseObject(models.Model):
+    name = models.CharField(max_length=63)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        abstract = True
+
+    @property
+    def db_name(self):
+        return db_name(self.name)
+
+    def __str__(self):
+        return f"{self.__class__.__name__}: {self.name}"
+
+
+class Collection(DatabaseObject):
+    schema = models.CharField(max_length=63)
+
+    @property
+    def db_collection(self):
+        return DBCollection(
+            self.db_name,
+            metadata,
+            schema=self.schema,
+            autoload_with=engine,
+            extend_existing=True,
+        )
```

Comment:
I'm worried that accessing the actual collection through the model will be a problem (or at least confusing).  It kind of muddles SQLAlchemy tables with Django ORM tables.  My preference would be to use the model strictly for mapping id strings to other id  strings, and keeping created/modified info if desired.

For example, using the Detail view is going to involve importing Django objects as well as SQLAlchemy.  I think these should be kept as cleanly separated as possible.

Reply:
@mathemancer the `db_collection` method just retrieves the table using the id string stored in the model. I think it's cleanest/the least amount of code to get the data related to a collection this way, and we can take advantage of Django features that hook into the ORM (like DetailViews), we don't want to reinvent things that Django provides for free.

I don't think we can get away from using both Django objects and SQLAlchemy objects to access data – if someone tries to look at a collection through the web UI or API, Django is the source of truth for the ID and SQLAlchemy is the source of truth for the actual data.